adapters/webhdfs-adapter/src/main/java/com/connexta/replication/adapters/webhdfs/WebHdfsNodeAdapter.java

```diff
@@ -0,0 +1,267 @@
+public class WebHdfsNodeAdapter implements NodeAdapter {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(WebHdfsNodeAdapter.class);
+
+  private final URL webHdfsUrl;
+
+  private final CloseableHttpClient client;
+
+  /**
+   * Adapter to interact with a Hadoop instance through the webHDFS REST API
+   *
+   * @param webHdfsUrl the address of the REST API for the
+   * @param client performs the HTTP requests
+   */
+  public WebHdfsNodeAdapter(URL webHdfsUrl, CloseableHttpClient client) {
+    this.webHdfsUrl = webHdfsUrl;
+    this.client = client;
+  }
+
+  @Override
+  public boolean isAvailable() {
+    LOGGER.debug("Checking access to: {}", webHdfsUrl);
+
+    try {
+      URIBuilder builder = new URIBuilder(webHdfsUrl.toString());
+      builder.setParameter("op", "CHECKACCESS").setParameter("fsaction", "ALL");
+
+      HttpGet httpGet = new HttpGet(builder.build());
+
+      ResponseHandler<Boolean> responseHandler =
+          response -> {
+            int status = response.getStatusLine().getStatusCode();
+
+            return status == HttpStatus.SC_OK;
+          };
+
+      return sendHttpRequest(httpGet, responseHandler);
+    } catch (URISyntaxException e) {
+      LOGGER.error("Unable to check availability, due to invalid URL.", e);
+      return false;
+    }
+  }
+
+  @Override
+  public String getSystemName() {
+    return "webHDFS";
+  }
+
+  @Override
+  public QueryResponse query(QueryRequest queryRequest) {
+    return null;
+  }
+
+  @Override
+  public boolean exists(Metadata metadata) {
+    return false;
+  }
+
+  @Override
+  public boolean createRequest(CreateRequest createRequest) {
+    return false;
+  }
+
+  @Override
+  public boolean updateRequest(UpdateRequest updateRequest) {
+    return false;
+  }
+
+  @Override
+  public boolean deleteRequest(DeleteRequest deleteRequest) {
+    return false;
+  }
+
+  @Override
+  public ResourceResponse readResource(ResourceRequest resourceRequest) {
+    return null;
+  }
+
+  @Override
+  public boolean createResource(CreateStorageRequest createStorageRequest) {
+    try {
+      String location = getLocation(createStorageRequest);
+
+      return writeFileToLocation(createStorageRequest, location);
+
+    } catch (URISyntaxException e) {
+      LOGGER.error("Unable to get location, due to invalid URL.", e);
+    } catch (ReplicationException e) {
+      LOGGER.error("Unable to create resource.", e);
+    }
+    return false;
+  }
+
+  /**
+   * Sends a PUT request with no file data and without following redirects, in order to retrieve the
+   * location to write to
+   *
+   * @param createStorageRequest request containing the {@link
+   *     org.codice.ditto.replication.api.data.Resource} to create
+   * @return a {@code String} containing the location to write to
+   * @throws URISyntaxException when the fileUrl cannot be parsed as a URI reference
+   */
+  @VisibleForTesting
```

Comment:
❓ Do you need this annotation since the method is already package private?

Reply:
The only reason this is package private is so that I can call the method directly from the unit tests.  While the annotation is purely informational, it is useful for anyone unfamiliar with the code looking at access levels.